packages/flutter/lib/src/rendering/animated_size.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/animation.dart';
+import 'package:meta/meta.dart';
+
+import 'box.dart';
+import 'object.dart';
+import 'shifted_box.dart';
+
+/// A render object that can render child size change animations. It works by
+/// always animated from its old child's size to the new one with the given
+/// duration and curve. In case the child changes its size in every single frame
+/// the object deduces that its size is animated from somehwere else, it
+/// skips/stops its animation and immediately changes its size to match the
+/// child.
+///
+/// In case the child overflows the current animated size of the parent, it gets
+/// clipped automatically.
+class RenderAnimatedSize extends RenderAligningShiftedBox {
+  /// Creates a render animated size. [duration] and [curve] define the
```

Comment:
"a render animated size" doesn't make sense. "Creates a render object that animates its size to match its child".

Don't start sentences with a lowercase letter (which means don't start sentences with a property name). Instead of "[foo] does bar", say "The [foo] argument does bar".